storage/consistent_watchable_store.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"encoding/binary"
+	"log"
+)
+
+var (
+	indexKeyName = []byte("index")
+)
+
+// IndexGetter is an interface that wraps the Get method.
+// Index is the indication of one txn. Each txn should have a unique index.
+type IndexGetter interface {
+	// Get gets the index of the on-going txn.
```

Comment:
If you mention txn in the interface, then you have to rename IndexGetter to TxnIndexGetter or something more specific.